shamie-Resnet.mdx

```diff
@@ -1,24 +1,126 @@
-Resnet Paper Summary:
+## Resnet Paper Summary
 
-Introduction:
-Observed Problem: Degradation Problem
-As the depth of the neural network deepens, the accuracy starts to saturate.
-
-The pre-existing perspectives of deeper layers was always thought of as effective because the performance of neural networks improved a lot by deepening the layers.
+### Observed Problem: Degradation with depth.
 
+Deeper network layers were assumed to be more effective because the performance of neural networks improved by deepening the layers.
 As the layers deepened, the extracted features could be further enriched, such as seen with VGG16 and VGG19.
+A question arose: "Is learning networks as easy as stacking more layers"?
+An obstacle to answering this question, the gradient vanishing problem, was addressed by normalized initializations and intermediate normalization layers.
+
+However, a new issue emerged: the degradation problem. As the neural networks deepened, accuracy saturated and degraded rapidly. An experiment comparing shallow and deep plain networks revealed that deeper models exhibited higher training and test errors, suggesting a fundamental challenge in training deeper architectures effectively. This degradation was not because of overfitting but because the training error increased when the network deepened. The added layers did not approximate the identity function.
+
+Resnet’s residual connections unlocked the potential of extreme depth, propelling accuracy upwards while plain networks grappled with degradation.
+
+## Resnet Architecture
+
+- A Residual Block. Source: ResNet Paper
+
+Resnet’s building blocks designed as identity functions, preserve input information while enabling learning. This approach ensures efficient weight optimization and prevents degradation as the network deepens.
+
+To make H approximate the identity function, we do the following
+
+### Let H(x) - x = f(x)
+
+### F(x) := 0
+
+### H(x) = 0 + x
+
+### H(x) = x
+
+Shortcut connections perform identity mapping and their output is added to the output of the stacked layers. Identity shortcut connections add neither extra parameters nor
+computational complexity, these connections bypass layers, creating direct paths for information flow, and they enable neural networks to learn the residual function (F).
+
+Resnet Network = Plain Network + Shortcuts
+
+For operations F(x) + x, F(x) and x should have identical dimensions.
+ResNet employs two techniques to achieve this:
+
+- Zero-padding shortcuts that add channels with zero values, maintaining dimensions without introducing extra parameters to be learned.
+- Projection shortcuts that use 1x1 convolutions to adjust dimensions when necessary, involving some additional learnable parameters.
+
+In deeper ResNet architectures like ResNet 50, 101, and 152, a specialized "bottleneck building block" is employed to manage parameter complexity and maintain efficiency while enabling even deeper learning.

```

Comment:
can you add one or two lines about deeper Resnet architectures ? i mean the problem with the deeper networks